### PR TITLE
feat: migrate Verdaccio.org to Docusaurus v2 and new design

### PR DESCRIPTION
**Pull request by semoal**
First of all, we should give thanks to @ploh007, for his amazing contribution gifting us this design.

## Pending to-do's (In order)

- [x] Integrate Crowdin, I already know how to do this, just need some conversations with Juan (Api keys, and some details..).
- [x] Integrate Netlify, probably we can do this together too Juan, I want to introduce a `netlify.toml`, to add a cache of Docusaurus build, since under the hood uses Webpack 5 it contains persistent caching, which means consequent builds will be a lot faster. Example: https://github.com/facebook/docusaurus/blob/master/website/netlify.toml

### Less relevant
- [x] Contact Algolia once released, to add the API_KEY for an amazing search experience.

### Demo

https://user-images.githubusercontent.com/22656541/126541663-4f52d550-ea19-4448-8f47-2b8a8f850d92.mp4

### CODE REVIEW

**Comment by juanpicado** [comment]
@semoal could you commit something small, I just enabled netlify preview for this branch.

**Comment by juanpicado** [comment]
Regarding Aligolia, we just need to PR here any future change

https://github.com/algolia/docsearch-configs/blob/master/configs/verdaccio.json

**Comment by semoal** [comment]
> Regarding Aligolia, we just need to PR here any future change
> 
> [https://github.com/algolia/docsearch-configs/blob/master/configs/verdaccio.json](https://github.com/algolia/docsearch-configs/blob/master/configs/verdaccio.json?rgh-link-date=2021-07-21T18%3A45%3A47Z)

They'll do it automatically for us, we just need to ping them

**Comment by semoal** [comment]
Well @juanpicado we're almost ready to 🚢.

### Crowdin
Crowdin as you could check it's correctly imported with the new layout of translations, everything is translated as previously, there were a lot of buggy markdown's, I had to review one file per one haha. Also I've had to create a tool (https://github.com/semoal/mdx-localized-headings) for migrating translated headings with their correspondent ids, because translated documents didn't worked fine with the links. Now, everything will run smoothly.

### Netlify

Also it's working perfectly fine, we should disable from the Netlify panel the auto build since we already do this operation. Also, I recommended changing the site name to verdaccio.netlify.com or something like that, in that way the deployment previews looks a bit nicer. 

### Known issues

- [ ] Safari IOS, circles inside Twitter Carrousel are not working properly.
- [ ] Safari IOS, lazy loaded iframes + twitter not working fine.

I'll look into this issues tomorrow early, in that way we can ship the website on Monday/Sunday whenever you want :)

**Comment by juanpicado** [comment]
Awesome !! I'll check today

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -6,12 +6,12 @@ const isDeployPreview = process.env.CONTEXT === 'deploy-preview';
 const i18nConfig = {
   defaultLocale: 'en',
   locales: isDeployPreview ? ['en'] : [
-    'ar',    'cs',    'de',    'en',
-    'es-ES', 'fil',   'fr',    'gl',
-    'hi',    'it',    'ja',    'ko',
-    'pl',    'pt-BR', 'ro',    'ru',
-    'sr',    'sr-CS', 'tg',    'vi',
-    'yo',    'zh-CN', 'zh-TW'
+    "ar-SA", "cs-CZ", "de-DE", "es-ES", "fil-PH",
```
Comment (juanpicado) [inline website/docusaurus.config.js:9]: I might disable some languages, I'll check which does not worth to enable it.

File: website/package.json [hunk 2]
```diff
@@ -1,18 +1,77 @@
...
+    "@docusaurus/module-type-aliases": "^2.0.0-beta.3",
+    "@tsconfig/docusaurus": "^1.0.2",
+    "@types/react": "^17.0.14",
+    "@types/react-helmet": "^6.1.2",
+    "@types/react-router-dom": "^5.1.8",
+    "@typescript-eslint/eslint-plugin": "^4.28.3",
+    "@typescript-eslint/parser": "^4.28.3",
+    "cross-env": "^7.0.3",
+    "docusaurus-plugin-sass": "^0.2.1",
+    "esbuild": "^0.12.15",
```
Comment (juanpicado) [inline website/package.json:64]: great

File: website/package.json [hunk 3]
```diff
@@ -1,18 +1,77 @@
...
+    "esbuild-loader": "^2.13.1",
+    "eslint": "^7.30.0",
```
Comment (juanpicado) [inline website/package.json:66]: We have these deps, eslint, prettier and typescript already on the root, but we can clean up that later.

File: website/package.json [hunk 4]
```diff
@@ -1,18 +1,77 @@
...
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
   },
-  "license": "MIT",
   "devDependencies": {
-    "docusaurus": "^1.14.7"
+    "@docusaurus/module-type-aliases": "^2.0.0-beta.3",
+    "@tsconfig/docusaurus": "^1.0.2",
+    "@types/react": "^17.0.14",
```
Comment (juanpicado) [inline website/package.json:57]: In theory, `@types/` should be on the root using pnpm workspaces, I'm not sure if the build is ignoring this and take those from the root. Not 100% sure.

File: package.json [hunk 5]
```diff
@@ -154,7 +155,10 @@
     "ci:version:changeset": "changeset version",
     "ci:publish": "changeset publish",
     "ts:ref": "update-ts-references --discardComments",
-    "website": "pnpm build --filter ...@verdaccio/website"
+    "website": "pnpm build --filter ...@verdaccio/website",
+    "crowdin:upload": "crowdin upload sources --auto-update --config ./crowdin.yaml",
+    "crowdin:download": "crowdin download --config ./crowdin.yaml",
```
Comment (juanpicado) [inline package.json:160]: do we need the secrets for this as well?

Comment (semoal) [reply package.json:160]: Yes, these two:
`CROWDIN_VERDACCIO_PROJECT_ID`
`CROWDIN_VERDACCIO_API_KEY`

File: website/package.json (hunk 4 above)
Comment (semoal) [reply website/package.json:57]: Didn't know that, thought that pnpm workspaces did hoisting like yarn workspaces/lerna...

File: website/docusaurus.config.js (hunk 1 above)
Comment (semoal) [reply website/docusaurus.config.js:9]: I think we should enable:

- es
- ch-tw
- cz
- fr
- de
- it
- po
- pt
- ru
- sr
- vi
- yo

File: website/package.json (hunk 3 above)
Comment (semoal) [reply website/package.json:66]: Done :)

File: website/package.json (hunk 4 above)
Comment (semoal) [reply website/package.json]: Done 👍

**Comment by semoal** [comment]
Screenshot 2021-07-26 at 14 13 44
Screenshot 2021-07-26 at 14 13 23

All the issues are fixed, we just need to add the Crowdin keys to secrets, and we're ready to go!

**Comment by juanpicado** [comment]
@semoal I had to downgrade `pnpm@6.10.3`, there is an issue with the next minor.

File: package.json (hunk 5 above)
Comment (juanpicado) [reply package.json:162]: done ✅

File: website/package.json (hunk 4 above)
Comment (juanpicado) [reply website/package.json]: I think it does, but on the next major is gone https://github.com/pnpm/pnpm/discussions/3536 so good move :)

File: website/docusaurus.config.js (hunk 1 above)
Comment (juanpicado) [reply website/docusaurus.config.js]: Agree with the list.

**Review by juanpicado — DISMISSED** [review]
LGTM 💯 great job, your contribution is huge 👏🏼 
Tested: mobile, local build, deployment, all good by far.

**Comment by semoal** [comment]
> LGTM 💯 great job, your contribution is huge 👏🏼 
> 
> Tested: mobile, local build, deployment, all good by far.

Thanks to you Juan. Looking forward to contribute even more

**Comment by juanpicado** [comment]
How do we solve the `API_KEY` for aligolia? Do I need to take an action here?

**Comment by semoal** [comment]
> How do we solve the `API_KEY` for aligolia? Do I need to take an action here?

Probably the current one will work, let me ask to Algolia team.

**Comment by semoal** [comment]
Ready to ship :shipit: 

shipit

**Review by juanpicado — APPROVED** [review]

**Pull request merged**